api/src/resolver_repo.ts

```diff
@@ -497,6 +497,68 @@ async function deletePods(_, { ids }: { ids: string[] }, { userId }) {
   return true;
 }
 
+async function copyRepo(_, { repoId }, { userId }) {
+  // Find the repo
+  const repo = await prisma.repo.findFirst({
+    where: {
+      id: repoId,
+    },
+    include: {
+      pods: {
+        include: {
+          parent: true,
+        },
+      },
+    },
+  });
+  if (!repo) throw new Error("Repo not found");
+
+  // Create a new repo
+  const { id } = await createRepo(_, {}, { userId });
+  // update the repo name
+  await prisma.repo.update({
+    where: {
+      id,
+    },
+    data: {
+      name: repo.name ? `Copy of ${repo.name}` : `Copy of ${repo.id}`,
+    },
+  });
+
+  // Create new id for each pod
+  const sourcePods = repo.pods;
+  const idMap = await sourcePods.reduce(async (acc, pod) => {
+    const map = await acc;
+    const newId = await nanoid();
+    map.set(pod.id, newId);
+    return map;
+  }, Promise.resolve(new Map()));
+
+  // Update the parent/child relationship with their new ids
+  const targetPods = sourcePods.map((pod) => {
+    return {
+      ...pod,
+      id: idMap.get(pod.id),
+      parent: pod.parent ? { id: idMap.get(pod.parent.id) } : undefined,
+      repoId: id,
+      parentId: pod.parentId ? idMap.get(pod.parentId) : undefined,
+    };
+  });
+
+  // Add all nodes without parent/child relationship to the new repo.
+  // TODO: it updates the parent/child relationship automatically somehow,maybe because the parentId? Try to figure out why, then refactor addPods method.
```

Comment:
Two comments:
1. The parent/children relationship is a **single** relation (named the `PARENT` relation in the schema). Thus, updating one should automatically update another. We should probably only update the `parent` field and not touch the `children` field at all.
2. Yep, providing the `parentId` field seems to be equivalent of using `{connect: {parent: {id}}}` Prisma clause.